### PR TITLE
[4.0] Remove Confirm Password from user edit view

### PR DESCRIPTION
**Pull request by wojsmol**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/23345 .

### Summary of Changes
Confirm Password field removed from user edit in backend
### Testing Instructions
code review or test whether user edit forks as expected and Confirm Password is not present.
### Expected result
No Confirm Password field and user edit works.
### Actual result
Confirm Password field exists and user edit works
### Documentation Changes Required
no

### CODE REVIEW

**Comment by chmst** [comment]
I see that the password is removed from the user edit view. 
But there are about 30 places where the password2 is used in the code. 
It is checked in the models and appears in different views and models in the frontend.

So I think it is too early for submitting a test result. 


<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23357">issues.joomla.org/tracker/joomla-cms/23357</a>.</sub>

**Comment by tonypartridge** [comment]
Why would you want to do this? For BC you should at least add it as a config option which is showing both by default.

**Comment by chmst** [comment]
The second password is only wasted time and no longer up to date in modern apps (my opinion). 
I do not insist but why then do we have the new password with the "plain text" button?  

If it is a B/C problem, we need a param and or a solution for the password2 check. But I do not see a big B/C problem. The password2 is not in the database. It could be used in extensions of the registration form, but I am sure that there we can find a solution.

File: libraries/src/User/User.php [hunk 1]
```diff
@@ -615,12 +605,6 @@ public function bind(&$array)
 			// Updating an existing user
 			if (!empty($array['password']))
 			{
-				if ($array['password'] != $array['password2'])
-				{
-					$this->setError(Text::_('JLIB_USER_ERROR_PASSWORD_NOT_MATCH'));
-
-					return false;
-				}
```
Comment (Quy) [inline libraries/src/User/User.php]:
```suggestion
```

File: libraries/src/User/User.php [hunk 2]
```diff
@@ -582,16 +582,6 @@ public function bind(&$array)
...
 			{
 				$array['password']  = UserHelper::genRandomPassword();
-				$array['password2'] = $array['password'];
-			}
-
-			// Not all controllers check the password, although they should.
-			// Hence this code is required:
-			if (isset($array['password2']) && $array['password'] != $array['password2'])
-			{
-				Factory::getApplication()->enqueueMessage(Text::_('JLIB_USER_ERROR_PASSWORD_NOT_MATCH'), 'error');
```
Comment (Quy) [inline libraries/src/User/User.php:592]: Remove language string.

**Comment by brianteeman** [comment]
I don't see the reason for this change

**Comment by dgrammatiko** [comment]
>  should at least add it as a config option

Please NO more switches...

**Comment by chmst** [comment]
I have tested and it looks great. But in the confirm.xml still is a password2. Could you remove this too?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23357">issues.joomla.org/tracker/joomla-cms/23357</a>.</sub>

**Comment by wojsmol** [comment]
@chmst Can you post a path to confirm.xml file in question?

**Comment by chmst** [comment]
It is components\com_users\forms\reset_complete.xml

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on a921c44bbd0aa2ac8ee8ff74fad431c5a7c0e5f1<br /><br />Thank you @wojsmol <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23357">issues.joomla.org/tracker/joomla-cms/23357</a>.</sub>

**Comment by tonypartridge** [comment]
Thinking about this, if we proceed with it... what about showing the password rather than hiding by default so people can see if it's typed correctly?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23357">issues.joomla.org/tracker/joomla-cms/23357</a>.</sub>

**Comment by dgrammatiko** [comment]
@tonypartridge the password shouldn't be visible by default and also in the field (password) was added a button to reveal the text just for this scenario. FWIW the field still **needs** to become a custom element, there used to be a PR by me and would be really nice if someone resurrects it because the current code is not production-ready...

**Comment by shinderohanoffical** [comment]
I have tested this item :red_circle: unsuccessfully on a921c44bbd0aa2ac8ee8ff74fad431c5a7c0e5f1<br /><br />The file marked for modification does not exist: administrator/components/com_admin/Model/ProfileModel.php				<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23357">issues.joomla.org/tracker/joomla-cms/23357</a>.</sub>

**Comment by idefax** [comment]
I have tested this item :red_circle: unsuccessfully on a921c44bbd0aa2ac8ee8ff74fad431c5a7c0e5f1<br /><br />Works not for me as well:

_The file marked for modification does not exist: administrator/components/com_admin/Model/ProfileModel.php_ 

PHP Version 	7.3.8
Web Server 	Apache
WebServer to PHP Interface 	cgi-fcgi
Joomla! Version 	Joomla! 4.0.0-beta1-dev Development [ Amani ] 17-October-2019 20:21 GMT
User Agent 	Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:69.0) Gecko/20100101 Firefox/69.0 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23357">issues.joomla.org/tracker/joomla-cms/23357</a>.</sub>

**Comment by brianteeman** [comment]
This really should be closed do people don't waste their time

**Pull request closed without merging**